### PR TITLE
[function] Add arc trigonometric function

### PR DESCRIPTION
**Pull request by Junnplus**
I hereby agree to the terms of the CLA available at: https://databend.rs/policies/cla/

## Summary

```
mysql> SELECT asin(0.2);
+--------------------+
| asin(0.2)          |
+--------------------+
| 0.2013579207903308 |
+--------------------+
1 row in set (0.02 sec)
Read 1 rows, 1 B in 0.003 sec., 290.12 rows/sec., 290.12 B/sec.

mysql> SELECT asin(1.1);
+-----------+
| asin(1.1) |
+-----------+
|      NULL |
+-----------+
1 row in set (0.01 sec)
Read 1 rows, 1 B in 0.002 sec., 594.19 rows/sec., 594.19 B/sec.

mysql> SELECT acos(1);
+---------+
| acos(1) |
+---------+
|       0 |
+---------+
1 row in set (0.01 sec)
Read 1 rows, 1 B in 0.002 sec., 489.93 rows/sec., 489.93 B/sec.

mysql> SELECT acos(1.0001);
+--------------+
| acos(1.0001) |
+--------------+
|         NULL |
+--------------+
1 row in set (0.02 sec)
Read 1 rows, 1 B in 0.003 sec., 321.45 rows/sec., 321.45 B/sec.

mysql> SELECT atan(-2);
+---------------------+
| atan((- 2))         |
+---------------------+
| -1.1071487177940906 |
+---------------------+
1 row in set (0.02 sec)
Read 1 rows, 1 B in 0.003 sec., 330.46 rows/sec., 330.46 B/sec.

mysql> SELECT atan(-2, 2);
+---------------------+
| atan((- 2), 2)      |
+---------------------+
| -0.7853981633974483 |
+---------------------+
1 row in set (0.01 sec)
Read 1 rows, 1 B in 0.002 sec., 587.31 rows/sec., 587.31 B/sec.

mysql> SELECT atan2(-2, 2);
+---------------------+
| atan2((- 2), 2)     |
+---------------------+
| -0.7853981633974483 |
+---------------------+
1 row in set (0.01 sec)
Read 1 rows, 1 B in 0.002 sec., 531.93 rows/sec., 531.93 B/sec.
```

## Changelog

- New Feature

## Related Issues

https://github.com/datafuselabs/databend/issues/2563

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

File: common/functions/src/scalars/maths/trigonometric.rs [hunk 1]
```diff
@@ -60,29 +69,83 @@ impl Function for TrigonometricFunction {
...
+                .to_minimal_array()?
+                .cast_with_type(&DataType::Float64)?;
+            let x_series = columns[1]
+                .column()
+                .to_minimal_array()?
+                .cast_with_type(&DataType::Float64)?;
+
+            let opt_iter = y_series
+                .f64()?
+                .into_no_null_iter()
```
Comment (sundy-li) [inline common/functions/src/scalars/maths/trigonometric.rs:144]: x, y may have different sizes.

X: Array[N]   --> to_minimal ---> array[N]
Y: Constant[N] --> to_minimal --> Constant[1]

Comment (Junnplus) [reply common/functions/src/scalars/maths/trigonometric.rs]: Thanks @sundy-li .

Update [here](https://github.com/datafuselabs/databend/pull/2888/files#diff-e4c1b93add37f16f9ffd647fd6da79377efef26c78eb8b47979636b3145c3c35R136-R163) for check x, y sizes .

File: common/functions/src/scalars/maths/trigonometric.rs [hunk 2]
```diff
@@ -60,29 +69,96 @@ impl Function for TrigonometricFunction {
...
+            let x_column: &DataColumn = &columns[1].column().cast_with_type(&DataType::Float64)?;
+
+            match (y_column, x_column) {
+                (DataColumn::Array(y_series), DataColumn::Constant(x, _)) => {
+                    let x = DFTryFrom::try_from(x.clone())?;
+                    let opt_iter = y_series.f64()?.into_iter().map(|y| y.map(|y| y.atan2(x)));
+                    DFFloat64Array::new_from_opt_iter(opt_iter)
+                }
+                (DataColumn::Constant(y, _), DataColumn::Array(x_series)) => {
+                    let y: f64 = DFTryFrom::try_from(y.clone())?;
```
Comment (sundy-li) [inline common/functions/src/scalars/maths/trigonometric.rs:143]: If `y` is None, the result should be None.

**Review by sundy-li — APPROVED** [review]

File: common/functions/src/scalars/maths/trigonometric.rs (hunk 2 above)
Comment (Junnplus) [reply common/functions/src/scalars/maths/trigonometric.rs]: You are right. I missed this. > <

**Comment by sundy-li** [comment]
@Junnplus thanks for the contributions.

**Pull request merged**